frontend/src/components/NewProject/ConflictModal.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useContext, useEffect } from 'react'
+import { Button, Form, Input, Modal } from 'semantic-ui-react'
+
+import UseForm from '@hooks/useForm'
+import ExistingProjectFromModel from '@models/ExistingProjectForm'
+import { isUsableProjectName } from '@utils/giteaApi'
+import { formatAsGiteaRepoName } from '@utils/index'
+import { AuthContext } from '@contexts/AuthContext'
+
+const ConflictModal = ({
+  conflictModalOpen,
+  onClose,
+  originalProjectName,
+  message,
+  onDifferentName,
+  onDifferentNameButtonContent,
+  onOverwrite,
+  onOverwriteButtonContent,
+}: BaseConflictModalProps) => {
```

Comment:
I'm not too familiar with the higher-order-component terminology but it is doesn't look like a HOC to me. A HOC from [React docs](https://reactjs.org/docs/higher-order-components.html) is something like:

```js
const EnhancedComponent = higherOrderComponent(WrappedComponent);
```

Reply:
I don't know what to call it :laughing: it\s what react calls [specialization](https://reactjs.org/docs/composition-vs-inheritance.html#specialization), maybe just call both of them components.